### PR TITLE
Issue #3458: Moved the message for enabling the color module into the…

### LINKED ISSUE #3458: [UX][DX] Improve the message about theme color.module support.
While reviewing #3257, I have noticed that when color.module is disabled, we show [a message in the theme settings page of Basis](https://github.com/backdrop/backdrop/blob/1.x/core/themes/basis/theme-settings.php#L11):

> This theme supports custom color palettes if the Color module is enabled on the [modules page](). Enable the Color module to customize this theme

...the link in that message points to `/admin/modules`.

Things we could improve:

1. The message could be a `backdrop_set_message()` of type `info` instead of a form `#markup` element.

2. The entire text could be shortened to:

    > This theme supports custom color palettes. [Enable the Color module]() to start customising it.

3. The link to the modules page could also be filtered to show only the color module (`/admin/modules?search=color`), and we should add destination so that the user is taken back to the theme settings page after enabling the module.

4. This bit here seems like code which could benefit all themes that support color.module. Perhaps consider moving the logic out of Basis and into a system-wide module.

5. When color.module is disabled, there are no other form elements left in that page (other than the `#markup` message), so the "Save theme settings" button does not make sense:

    screen shot 2019-01-04 at 3 12 21 am

    ...we should have some logic in place so that this submit button is rendered only if there are settings to save.

---
PR by @hosef: https://github.com/backdrop/backdrop/pull/2614

### PR DESCRIPTION
**Pull request by hosef**
… system module so that themes are no longer responsible for it, improved the message, and hid the submit button when there are no settings.

Fixes https://github.com/backdrop/backdrop-issues/issues/3458

### CODE REVIEW

**Comment by jenlampton** [comment]
Closing for new sandbox.

**Comment by jenlampton** [comment]
Re-opening for new sandbox

**Comment by herbdool** [comment]
Reopening the sandbox

File: core/modules/system/system.admin.inc [hunk 1]
```diff
@@ -423,6 +423,8 @@ function system_theme_settings($form, &$form_state, $key) {
     '#type' => 'value',
     '#value' => $key,
   );
+  // Store the form before processing theme settings so we can check for changes.
```
Comment (klonos) [inline core/modules/system/system.admin.inc:426]:
```suggestion
  // Store the form before processing theme settings, so we can check for
  // changes.
```

**Comment by laryn** [comment]
Thanks for the PR @hosef and for the review @klonos! Merged into 1.x and 1.19.x.

**Comment by jenlampton** [comment]
This PR also incldued a change to the submit button that breaks contrib modules that use this form (since the form can't be saved without the submit button). PR that restores the submit button posted at https://github.com/backdrop/backdrop-issues/issues/5308

**Pull request merged**